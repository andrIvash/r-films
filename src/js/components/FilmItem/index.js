// @flow
import React from 'react';
import type { Film } from '../../flow-types.js';

type Props = {
  film: Film,
  id: number,
  onFilmSelect: (id: number) => void,
}

const FilmItem = (props:Props) => {
  const { film, onFilmSelect, id } = props;

  return (
    <li className='film-list__item' onClick={() => {onFilmSelect(id)}}>
      <div className='film-item'>
        <div className='film-item__img-wrap'>
          <img alt='film' className='film-item__img' src={film.poster_path} />
        </div>
        <div className='film-item__info'>
          <div className='film-item__title'>{film.title}</div>
          <div className='film-item__year'>{film.release_date.substr(0,4)}</div>
        </div>
        <div className='film-item__genre'>{film.genres[0]}</div>
      </div>
    </li>
  );
};

export default FilmItem;
