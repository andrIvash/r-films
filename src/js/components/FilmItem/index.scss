.film-item {
  width: rem(225px);
  min-height: rem(380px);
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  cursor: pointer;
}

.film-item__img-wrap {
  width: rem(225px);
  height: rem(300px);
  margin-bottom: rem(10px);
  
  .film-item__img {
    width: 100%;
    height: 100%;
  }
}

.film-item__info {
  display: flex;
  width: 100%;
  max-height: rem(50px);
  overflow: hidden;
  justify-content: space-between;
  align-items: flex-start;
  font-family: $sign-font;
  margin-bottom: rem(10px);

}

.film-item__title {
  width: 75%;
  font-weight: 700;
  font-size: rem(17px);
  text-transform: uppercase;
}

.film-item__year {
  width: 20%;
  display: flex;
  align-content: center;
  justify-content: center;
  border: 1px solid $color-background;
  font-weight: 500;
  font-size: rem(14px);
}

.film-item__genre {
  width: 100%;
  display: flex;
  align-content: center;
  justify-content: flex-start;
  flex-wrap: wrap;
  font-family: $sign-font;
  font-weight: 700;
  font-size: rem(12px);
}