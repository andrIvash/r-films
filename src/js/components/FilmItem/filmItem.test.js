import React from 'react';
import FilmItem from './index';
import renderer from 'react-test-renderer';

let props = {};

jest.mock('react-router-dom/Link', () => 'router-link');
jest.mock('react-router-dom/NavLink', () => 'nav-link');

describe('FilmItem', () => {
  beforeEach(()=> {
    props = {
      film: {
        poster_path: 'http://test',
        release_date: '1924',
        title: 'title',
        genres: ['drama'],
      },
      id: 22,
      onFilmSelect: jest.fn(),
    };
  });
  it('should be defined', () => {
    expect(FilmItem).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = renderer.create(
      <FilmItem {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render write values ', () => {
    const wrapper = shallow(
      <FilmItem {...props} /> );
    expect(wrapper.find('.film-item__img')
      .getElement().props.src).toBe('http://test');
    expect(wrapper.find('.film-item__title').text()).toBe('title');
    expect(wrapper.find('.film-item__year').text()).toBe('1924');
  });

  it(`should calls onFilmSelect with an argument 
    when FilmItem is clicked`, () => {
    const wrapper = shallow(
      <FilmItem {...props} /> );
    wrapper.simulate('click');
    expect(props.onFilmSelect).toHaveBeenCalledTimes(1);
    expect(props.onFilmSelect).toHaveBeenCalledWith(22);
  });
});
