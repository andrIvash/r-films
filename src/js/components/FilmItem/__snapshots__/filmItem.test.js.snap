// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmItem should render correctly 1`] = `
<li
  class="film-list__item"
>
  <div
    mock="Link_component_mock"
  >
    <div
      class="film-item"
    >
      <div
        class="film-item__img-wrap"
      >
        <img
          alt="film"
          class="film-item__img"
          src="http://test"
        />
      </div>
      <div
        class="film-item__info"
      >
        <div
          class="film-item__title"
        >
          title
        </div>
        <div
          class="film-item__year"
        >
          1924
        </div>
      </div>
      <div
        class="film-item__genre"
      >
        drama
      </div>
    </div>
  </div>
</li>
`;
